File: fixtures/scale/objc/large/Families/Autokey/AutokeyDescriptor.m
#import "AutokeyDescriptor.h"
#import "AutokeyCipher.h"
#import "AutokeyVectors.h"

@implementation AutokeyDescriptor

- (NSString *)family {
    return @"autokey";
}

- (NSString *)suite {
    return @"polyalphabetic";
}

- (id<CLCipher>)cipher {
    return [[AutokeyCipher alloc] init];
}

- (NSArray *)vectors {
    return [AutokeyVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Autokey/AutokeyDescriptor.php
<?php

namespace CipherLab\Families\Autokey;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together AutokeyCipher and
// AutokeyVectors under the "polyalphabetic" suite.
class AutokeyDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'autokey';
    }

    public function suite(): string
    {
        return 'polyalphabetic';
    }

    public function cipher(): Cipher
    {
        return new AutokeyCipher();
    }

    public function vectors(): array
    {
        return AutokeyVectors::all();
    }
}
